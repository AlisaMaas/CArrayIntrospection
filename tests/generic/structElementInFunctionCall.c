#include "structHeader.h"
int find(char *string, char dest) {
    for (int i = 0; string[i]; i++) {
        if (dest == string[i]) {
            return 1;
        }
    }
    return 0;
}

void call(struct foo *f, char c) {
    find(f->string, c);
}