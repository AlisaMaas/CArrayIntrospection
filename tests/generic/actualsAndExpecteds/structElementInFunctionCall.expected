Printing analysis 'Promote Memory to Register' for function 'find':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Promote Memory to Register' for function 'call':
Pass::print not implemented for pass: 'Promote Memory to Register'!
Printing analysis 'Determine whether and how to annotate each function with the null-terminated annotation':
find with argument 0 should be annotated NULL_TERMINATED (2)  because found a non optional sentinel check for string in find
struct struct.foo[0] should be annotated NULL_TERMINATED (2) because  found a call to find passing string.
